core/src/duplicate_repair_status.rs

```diff
@@ -481,19 +465,20 @@ impl AncestorRequestStatus {
             if earliest_erroring_ancestor_index == agreed_response.len() - 1 {
                 // If the earliest ancestor is missing or a mismatch, then we need to keep searching
                 // for earlier mismatches
-                let repair_status = DuplicateSlotRepairStatus::new(agreed_response);
+                let repair_status =
+                    DuplicateSlotRepairStatus::new(*agreed_response.last().unwrap());
                 DuplicateAncestorDecision::ContinueSearch(repair_status)
             } else {
-                // We only need to look through the first `earliest_erroring_ancestor_index + 1`
-                // elements and dump/repair any mismatches.
-                agreed_response.truncate(earliest_erroring_ancestor_index + 1);
+                // We only need to dump and repair the earliest mismatching ancestor.
                 let repair_status = decision.repair_status_mut().unwrap();
-                repair_status.correct_ancestors_to_repair = agreed_response;
+                repair_status.correct_ancestor_to_repair =
+                    agreed_response[earliest_erroring_ancestor_index];
```

Comment:
This is the crux of the change to only send the earliest mismatched slot right?

Reply:
yeah exactly, only sends the earliest mismatch. in case there are mismatches after that the logic in `dump_then_repair` will remove descendants with frozen hashes